test/integration/targets/get_certificate/tests/validate.yml

```diff
@@ -0,0 +1,90 @@
+---
+- name: Test get_certificate
+  connection: local
+  hosts: localhost
+  gather_facts: false
+  tasks:
+
+    - name: Get servers certificate
+      get_certificate:
+        host: "{{ httpbin_host }}"
+        port: 443
+      register: result
+
+    - debug: var=result
+
+    - assert:
+        that:
+          # This module should never change anything
+          - result is not changed
+          - result is not failed
+          # We got the correct ST from the cert
+          - "'North Carolina' == result.cert.ST"
+
+    - name: Connect to http port (will fail because there is no SSL cert to get)
+      get_certificate:
+        host: "{{ httpbin_host }}"
+        port: 80
+      register: result
+      ignore_errors: true
+
+    - assert:
+        that:
+          - result is not changed
+          - result is failed
+          # We got the expected error message
+          - "'The handshake operation timed out' in result.msg"
+ 
+    - name: Test timeout option
+      get_certificate:
+        host: "{{ httpbin_host }}"
+        port: 1234
+        timeout: 1
+      register: result
+      ignore_errors: true
+
+    - assert:
+        that:
+          - result is not changed
+          - result is failed
+          # We got the expected error message
+          - "'Failed to get cert from port with error: timed out' == result.msg"
+
+    - name: Test failure if ca_certs is not a valid file
+      get_certificate:
+        host: "{{ httpbin_host }}"
+        port: 443
+        ca_certs: dn.e
+      register: result
+      ignore_errors: true
+
+    - assert:
+        that:
+          - result is not changed
+          - result is failed
+          # We got the correct response from the module
+          - "'ca_certs file does not exist' == result.msg"
+
+    - name: Get servers certificate comparing it to its own ca_cert file
+      get_certificate:
+        ca_certs: '{{ output_dir }}/temp.pem'
+        host: "{{ httpbin_host }}"
+        port: 443
+      register: result
+
+    - assert:
+        that:
+          - result is not changed
+          - result is not failed
+
+    - name: Get servers certificate comparing it to an invalid ca_cert file
+      get_certificate:
+        ca_certs: 'bogus_ca.pem'
+        host: "{{ httpbin_host }}"
+        port: 443
+      register: result
```

Comment:
If this task is supposed to fail, you need `ignore_errors: true`, otherwise the test will fail. (The test is a regular playbook.)